### ISSUE TITLE
Use environment markers in `setup.py` rather than explicitly switching requirements.

### ISSUE THREAD

**Issue by lovelydinosaur** [Kludex/uvicorn#135]
Recommended by @uranusjr, here: https://github.com/encode/uvicorn/issues/82#issuecomment-406148569

> I would strongly recommend using PEP 508 environment markers to specify dependencies instead. This allows for the same effect (in relatively recent Setuptools and pip), prevents the packager from needing to maintain platform-specific artifacts (e.g. wheels), and has immensely better compatibility with the new trend of locking-aware dependency management tools like Pipenv and Poetry.

> The following should be functionality-wise equivalent to the if-elif-else clause:

```python
requirements = [
    'click',
    'h11',
    'websockets>=6.0',
    "httptools; platform_system != 'Windows' and 'CYGWIN' not in platform_system and platform_python_implementation != 'PyPy'",
    "uvloop; platform_system != 'Windows' and 'CYGWIN' not in platform_system and platform_python_implementation != 'PyPy'",
]
```

**Comment by lovelydinosaur** [comment]
This would be a nice simple one for a contributor to jump in on! 😄

**Comment by lovelydinosaur** [comment]
Closing in favor of using optional dependencies #219

**Comment by vlcinsky** [comment]
PR #309 resolves this issue, released in uvicorn==0.4.6

This will change in next major version, where use of extras is to be implemented.
